### PR TITLE
Re-implement subsurface scattering.

### PR DESCRIPTION
**Pull request by reduz**
The size settings are more "just works", with default scale and depth scale
values that don't need much tweaking.

Additionally, a "skin" mode was added so skin looks better.

### CODE REVIEW

File: servers/rendering/rasterizer_rd/rasterizer_effects_rd.cpp [hunk 1]
```diff
@@ -1356,6 +1402,22 @@ RasterizerEffectsRD::RasterizerEffectsRD() {
 		ssr_scale.pipeline = RD::get_singleton()->compute_pipeline_create(ssr_scale.shader.version_get_shader(ssr_scale.shader_version, 0));
 	}
 
+	{
+		Vector<String> sss_modes;
+		sss_modes.push_back("\n#define USE_11_SAMPLES\n");
+		sss_modes.push_back("\n#define USE_17_SAMPLES\n");
+		sss_modes.push_back("\n#define USE_25_SAMPLES\n");
+		//sss_modes.push_back("\n#define VERTICAL_PASS\n");
```
Comment (clayjohn) [inline servers/rendering/rasterizer_rd/rasterizer_effects_rd.cpp:1410]: Should be deleted

File: servers/rendering/rasterizer_rd/rasterizer_effects_rd.cpp [hunk 2]
```diff
@@ -1356,6 +1402,22 @@ RasterizerEffectsRD::RasterizerEffectsRD() {
...
+
+		sss.shader.initialize(sss_modes);
+
+		sss.shader_version = sss.shader.version_create();
```
Comment (clayjohn) [inline servers/rendering/rasterizer_rd/rasterizer_effects_rd.cpp:1414]: Shader is created, but never freed. Should be freed in the destruction function. Should also add code to free the SSR shaders while you are at it.

**Comment by reduz** [comment]
@akien-mga : Concerns from @clayjohn solved, also cleaned up SSR filter shader a bit.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**